### DISCUSSION TITLE
Injeção de dependência

### DISCUSSION THREAD

**Discussion by juliocesarlab** [balta-io/forum#18] [Tira Dúvidas] [answered]
No curso de injeção de dependência é dito que ao usar o AddTransient, será criada uma nova instância para cada vez que a dependência for chamada. Isso significa que dentro de um mesmo método, toda vez que eu utilizar a dependência, eu terei uma nova instância desse item ? Ou seria uma instância para cada construtor que a dependência é "invocada" ?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @juliocesarlab, como vai?

Depende de onde você resolveu a dependência... Se resolveu ela no construtor do controller, vai ser criada uma instância lá e nos métodos abaixo deste, será utilizada a mesma instância.

Se resolveu direto no método, utilizando [FromServices], será uma instância para aquele método.

Tudo isto se anula se você utilizar o `new` e fizer uma nova instância da classe... aí independente do modelo que utilize, terá sempre uma nova instância.

**Comment by juliocesarlab** [comment]
Bom dia, obrigado pela resposta !
